bigbluebutton-html5/imports/startup/server/index.js

```diff
@@ -21,17 +21,35 @@ WebApp.connectHandlers.use('/check', (req, res) => {
 
 WebApp.connectHandlers.use('/locale', (req, res) => {
   const APP_CONFIG = Meteor.settings.public.app;
-  const defaultLocale = APP_CONFIG.defaultSettings.application.locale;
-  const localeRegion = req.query.locale.split(/[-_]/g);
-  const localeList = [defaultLocale, localeRegion[0]];
-
-  let normalizedLocale = localeRegion[0];
+  const fallback = APP_CONFIG.defaultSettings.application.fallbackLocale;
+  const browserLocale = req.query.locale.split(/[-_]/g);
+
+  const localeList = [fallback];
+  const getAvailableLocales = fs.readdirSync('assets/app/locales');
+  let regionDefault = null;
+  const usableLocales = [];
+
+  getAvailableLocales
+    .map(file => file.replace('.json', ''))
+    .map(locale => (
+      locale
+    ))
+    .reduce((i, locale) => {
```

Comment:
Hmm, is this correct? With this code you are using a reduce as a for-loop and not as a reducer.

Reply:
```